Sources/Linq2DynamoDb.DataContext/IEntityWrapper.cs

```diff
@@ -7,5 +7,6 @@ public interface IEntityWrapper
         object Entity { get; }
         Document GetDocumentIfDirty();
         void Commit();
+        Document AsDocument();
```

Comment:
For optimistic delete you also would need only keys and version field value, so no need in converting the whole entity even in that case.